Makefile

```diff
@@ -14,46 +14,135 @@
 # limitations under the License.
 #
 
+SHELL = /bin/bash
+
 ALL_VARIANTS = $(shell find live-build/variants -maxdepth 1 -mindepth 1 -exec basename {} \;)
 ALL_INTERNAL = $(shell find live-build/variants -maxdepth 1 -mindepth 1 -name 'internal-*' -exec basename {} \;)
 ALL_EXTERNAL = $(shell find live-build/variants -maxdepth 1 -mindepth 1 -name 'external-*' -exec basename {} \;)
 
+ALL_HYPERVISORS := generic aws azure gcp kvm
```

Comment:
I was confused by the term "generic" in the linux-pkg repo, and I think I'm still confused. What is a generic hypervisor? Where does VMware ESX fit in? We only need the VMware open-vm-tools package installed on that one, but we currently install it on all platforms as part of a `delphix-platform` dependency. That probably needs to be moved here as part of a VMware-specific build (?)...

Reply:
The `walinuxagent` package is similarly installed everywhere using a general dependency from `delphix-platform`, and it should only be installed on Azure.

Reply:
Ok, I guess I'm confused with the use of the term "generic" to describe a hypervisor (the variable is called `ALL_HYPERVISORS`) if there is no generic hypervisor. Anyway, my overall point/questio is: how do we build an ESX-specific variant (or whatever we decide to call that) with `open-vm-tools` installed?

Reply:
@sebroy Using the terminology I've presented in the post above, we'd build for the ESX `platform` by customizing the `delphix-platform` package. The idea is that there would be a different `delphix-platform` package for each `platform`. When appliance-build builds for a particular platform, it will select the appropriate `delphix-platform` package.

So in other words, most of that work would go into the `delphix-platform` project.